src/Symfony/Component/Yaml/Parser.php

```diff
@@ -237,7 +237,12 @@ private function doParse($value, $flags)
                 }
 
                 if (!(Yaml::PARSE_KEYS_AS_STRINGS & $flags) && !is_string($key)) {
-                    @trigger_error('Implicit casting of incompatible mapping keys to strings is deprecated since version 3.3 and will throw \Symfony\Component\Yaml\Exception\ParseException in 4.0. Pass the PARSE_KEYS_AS_STRING flag to explicitly enable the type casts.', E_USER_DEPRECATED);
+                    $keyType = is_numeric($key) ? 'numeric key' : 'incompatible key type';
+                    @trigger_error(sprintf(
+                        'Implicit casting of %s to string on line %d is deprecated since version 3.3 and will throw \Symfony\Component\Yaml\Exception\ParseException in 4.0. Pass the PARSE_KEYS_AS_STRING flag to explicitly enable the type casts.',
```

Comment:
please add double quotes around the first "%s"
please keep the notice on a single line also